### ISSUE TITLE
locale failure during installation wizard

### ISSUE THREAD

**Issue by half-monkey** [nextcloud/server#27018]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Follow instructions to install server on Linux here: https://docs.nextcloud.com/server/latest/admin_manual/installation/source_installation.html# on a Debian Bullseye installation that has some but not all of the locales in the error message.
2. When first visiting https://blahblahblah/nextcloud encounter error message: "

    Setting locale to en_US.UTF-8/fr_FR.UTF-8/es_ES.UTF-8/de_DE.UTF-8/ru_RU.UTF-8/pt_BR.UTF-8/it_IT.UTF-8/ja_JP.UTF-8/zh_CN.UTF-8 failed

    Please install one of these locales on your system and restart your webserver.
"

Please note I was able to unblock the installation by editing lib/private/legacy/OC_Util.php and forcing "isSetLocaleWorking" to return "true". Thereafter the installation worked as expected. I have at least de_DE.UTF-8 from the list above configured (en_GB.UTF-8 is my system default)

### Expected behaviour
Tell us what should happen
I expect to be asked to configure an admin account and database connection at this point.

### Actual behaviour
Tell us what happens instead
I get the error message:

    Setting locale to en_US.UTF-8/fr_FR.UTF-8/es_ES.UTF-8/de_DE.UTF-8/ru_RU.UTF-8/pt_BR.UTF-8/it_IT.UTF-8/ja_JP.UTF-8/zh_CN.UTF-8 failed

    Please install one of these locales on your system and restart your webserver.

### Server configuration

**Operating system:**
Debian Bullseye

**Web server:**
Apache2
**Database:**
MariaDB 10.5

**PHP version:**
7.4

**Nextcloud version:** (see Nextcloud admin page)
21.0

**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh install

**Where did you install Nextcloud from:**
https://nextcloud.com/install/#

**Signing status:**
LetsEncrypt certificate

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

Cannot log in as admin - unable to complete installation when bug manifests
```
</details>

**List of activated apps:**
None - unable to complete installation

**Comment by kesselb** [comment]
Duplicate of https://github.com/nextcloud/server/issues/12318
